### ISSUE TITLE
Allow spotless check/apply to run early in every build

### ISSUE THREAD

**Issue by niloc132** [deephaven/deephaven-core#2487]
As a dhc developer, I want to not have my CI builds fail from things that it should be able to automatically fix. Note that not all developers might be interested in this.

Some possible fixes:
 * Github Actions could create commits automatically - this would let a developer remove/revert the commit and push their own resolution. Cons: I'm not yet sure how to make a CI step create a new commit and push to a contributor's branch, appropriate error handling if this doesn't work.
 * Gradle could run apply or check as an early part of the build. This could be optional (and probably default to "off"), so that developers could decide to use either mode here ("fail my code if it doesn't conform" or "automatically make my code conform, I don't care about formatting specifics"). Cons, costs a small amount of time in front of every compile/run/test, though at least in theory this should be correctly cached by gradle.

**Comment by niloc132** [comment]
Note, this will also apply to copyright headers when I implement that (soon).

**Comment by niloc132** [comment]
Intellij's format is not the same as the formatting we are validating in the build - I believe that @devinrsmith looked for something that would align, but didn't find anything.

**Comment by devinrsmith** [comment]
I *think* trying to run spotlessApply as part of the task tree with downstream compiles may be problematic - at the very least, it would need to be the very first thing run b/c it potentially effects downstream inputs. I think having an ability to opt into git hooks pre-push could be useful - specifically, running `./gradlew spotlessCheck`.

**Comment by niloc132** [comment]
Yeah, it wouldn't be perfect, but could be hacked together - as an ugly example, afterEvaluate collect all tasks and make them all dependsOn the `spotlessCheck` (and, eventually, `license`) task. No codegen could sneak in before it, no compile could get into a weird state with seeing changes both before and after, etc.
